packages/core/components/other/nav-toggle/NavToggle.ts

```diff
@@ -1,162 +1,151 @@
-import { bind, Component, dataParam, emitEvent, Logger, register } from 'ovee.js';
+import { computed, defineComponent, emitEvent, Logger, useDataAttr } from 'ovee.js';
 
 type toggleEvent = CustomEvent<boolean>;
 
 const logger = new Logger('NavToggle');
 const DEFAULT_NAV_NAME = 'menu';
 
-@register('nav-toggle')
-export class NavToggle extends Component {
-	nav: HTMLElement;
-	animStarted: boolean;
-	html = document.documentElement;
+export const NavToggle = defineComponent((_, { on, emit }) => {
```

Comment:
to co bym najpierw zrobił to poukładał zmienne, na góre dałbym zmienne typu `html`, później `ref`-y, następnie `computed`-y i(useDataAttr itd.)

```ts
const html = document.documentElement;
// ...

const isOpen = ref(false);
// ...

const navName = useDataAttr('navToggle');
// ...

const someComputed = computed(() => {});
// ...

onMounted(() => {});
// ...

function someFn() {}
// ...

return {}
```